huxley/www/js/components/ChairSummaryView.js

```diff
@@ -0,0 +1,185 @@
+              <thead>
+                <tr>
+                  <th>Assignment</th>
+                  <th>Summary</th>
+                </tr>
+              </thead>
+              <tbody>
+                {this.renderSummaryRows()}
+              </tbody>
+            </table>
+          </div>
+          <Button
+            color="green"
+            onClick={this._handleSaveSummaries}>
+            Save
+          </Button>
+          <Button
+            color="blue"
+            onClick={this._handlePublishSummaries}>
+            Publish
+          </Button>
+        </form>
+      </InnerView>
+    );
+  },
+
+  renderSummaryRows() {
+    var committeeCountries = Object.keys(this.state.country_assignments);
+    var countries = this.state.countries;
+    return committeeCountries.map(country => {
+      var country_delegates = this.state.country_assignments[country];
+      return (
+        <tr key={country}>
+          <td>
+            {Object.keys(countries).length ? countries[country].name : country}
+          </td>
+          <td>
+            <textarea
+              className="text-input"
+              style={{"width": "95%"}}
+              rows="3"
+              onChange={this._handleSummaryChange.bind(this, country)}
+              defaultValue={country_delegates[0].summary}
+            />
+          </td>
+        </tr>
+      );
+    });
+  },
+
+  _handleSummaryChange(country, event) {
+    var newSummary = event.target.value;
+    var country_assignments = this.state.country_assignments;
+    var country_delegates = country_assignments[country];
+    for (var delegate of country_delegates) {
+      delegate.summary = newSummary;
+    }
+    this.setState({country_assignments: country_assignments});
+  },
+
+  _handleMapAssignments() {
+    var user = CurrentUserStore.getCurrentUser();
+    var country_assignments = {};
+    var delegates = this.state.delegates;
+    var assignments = this.state.assignments;
+    for (var delegate of delegates) {
+      var assignment = assignments.find(assignment => assignment.id == delegate.assignment);
+      if (!assignment) continue;
+
+      var countryID = assignment.country;
+      if (countryID in country_assignments) {
+        country_assignments[countryID].push(delegate);
+        continue;
+      }
+      
+      country_assignments[countryID] = [delegate];
+    }
+
+    this.setState({country_assignments: country_assignments});
+  }, 
+
+  _handleSaveSummaries(event) {
+    var committee = CurrentUserStore.getCurrentUser().committee;
+    var country_assignments = this.state.country_assignments;
+    var delegates = [];
+    for (var country in country_assignments) {
+      delegates = delegates.concat(country_assignments[country]);
+    }
+    DelegateActions.updateCommitteeDelegates(committee, delegates);
+  },
+
+  _handlePublishSummaries(event) {
+    var confirm = window.confirm("Advisors will be able to see everything you have entered. Do you wish to continue?");
```

Comment:
I'd say let's be even more explicit here. Something like "By pressing ok, you are allowing advisors to read the summaries that you have written about their delegations. Please ensure there are no inappropriate comments or language in any of these summaries. If there is none and you are ready to publish your summaries to advisors, press 'ok' to continue."

I hate to be a stickler about language, but I want to make sure there is absolutely no way for any of our chairs to accidentally publish anything that our advisors would find inappropriate.